ci/ci/github.py

```diff
@@ -362,11 +365,14 @@ def github_status(self):
                     self.source_sha_failed = True
                 self.target_branch.state_changed = True
 
-    async def _heal(self, batch_client, on_deck):
+    async def _heal(self, batch_client, pool, on_deck):
         # can't merge target if we don't know what it is
         if self.target_branch.sha is None:
             return
 
+        if not await self.authorized(pool):
+            return
```

Comment:
looks good, this should always protect the only call to `_start_build`. We gotta be careful in future changes that we never add other paths to `_start_build`.